modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDynamicParameterTest.java

```diff
@@ -482,6 +484,70 @@ public void testPrepareTimeout() {
         });
     }
 
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("decimalOverflows")
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testDecimalOpOverflow(String expr, Object param) {
+        assertThrowsSqlException(RUNTIME_ERR, "Numeric field overflow", () -> sql(expr, param));
+    }
+
+    private static Stream<Arguments> decimalOverflows() {
+        return Stream.of(
+                arguments("SELECT (? + 1)::DECIMAL(2, 0)", 99),
+                arguments("SELECT (? - 1)::DECIMAL(2, 0)", -99),
+                arguments("SELECT (? * 2)::DECIMAL(2, 0)", 50),
+                arguments("SELECT (? / 0.1)::DECIMAL(2, 0)", 10),
+
+                arguments("SELECT (9 + ?)::DECIMAL(2, 1)", 1),
+                arguments("SELECT (9 - ?)::DECIMAL(2, 1)", 19),
+                arguments("SELECT (9 * ?)::DECIMAL(2, 1)", 2),
+                arguments("SELECT (9 / ?)::DECIMAL(2, 1)", 0.5)
+        );
+    }
```

Comment:
may you elaborate, what exactly these tests cases are supposed to validate?

Reply:
this tests were requested from previous reviewer. I think they are not related to this issue too.

Reply:
we don't expect to have `Numeric field overflow` for arithmetic with decimals, thus I see no value in test complication. @ygerzhedovich does it make sense to you?

Reply:
It has value. After the test, I see that for some reason we have different messages for overflow decimal and another numerics - why? should we have the same message?
Also, we check all numerics except DECIMAL, why? All these tests are related to overflow during casts.

Do you agree with such an explanation?

Reply:
> After the test, I see that for some reason we have different messages for overflow decimal and another numerics - why?

is this difference caused by arithmetic? I don't think so

> Also, we check all numerics except DECIMAL, why?
because we don't expect to have `Numeric field overflow` for arithmetic with decimals. That's how it defined by our type system.

Reply:
ok, I was wrong. 
Sorry.

Reply:
I remove it